src/reflect/scala/reflect/internal/util/SourceFile.scala

```diff
@@ -187,6 +190,12 @@ class BatchSourceFile(val file : AbstractFile, content0: Array[Char]) extends So
     lastLine
   }
 
+  override def lines: Iterator[String] =
```

Comment:
for this use case line(i:Int) would be more efficient
or lines(start, end) 

I would be concerned leaving this in the common API as it as it with churn memory, e.g. in the way it is used.

Reply:
Agree, let's add `(start: Int = 0, end: Int = lineIndices.length)` parameters

Reply:
Do you mean returning a `String` or an `Iterator[String]`? I can't imagine (I'll check) that `lines(a, b).mkString` in the latter case would be too different than `(lines take a drop b).mkString`; the iterator is lazy so it will only make strings for the given lines. I'll change it to the former; we only need it in this context anyways.

Reply:
Dropped elements are still evaluated:
```
scala> (0 to 10).iterator.map(x => {println(x); x}).drop(3).take(3).toList
0
1
2
3
4
5
res0: List[Int] = List(3, 4, 5)
```

Reply:
... of course. Sorry, it's early here :P

Reply:
☕️  🥐